src/mdns.rs

```diff
@@ -88,6 +88,17 @@ impl From<mdns_result_t> for QueryResult {
             p = unsafe { (*p).next };
         }
 
+        #[cfg(esp_idf_version_major = "5")]
+        let interface = match unsafe { CStr::from_ptr(esp_netif_get_desc(result.esp_netif)) }
+            .to_bytes_with_nul()
+        {
+            b"sta\0" => Interface::STA,
+            b"ap\0" => Interface::AP,
+            b"eth\0" => Interface::ETH,
+            _ => unimplemented!(),
```

Comment:
This is not very elegant, as the user can give custom descriptions to `esp_netif`, and also there are multiple new `netif` types that aren't covered by these.

Reply:
Very shaky. I would put a TODO here even though we won't fix it right now.